src/main/kotlin/no/kartverket/matrikkel/bygning/matrikkel/adapters/MatrikkelBygningClient.kt

```diff
@@ -0,0 +1,50 @@
+package no.kartverket.matrikkel.bygning.matrikkel.adapters
+
+import no.kartverket.matrikkel.bygning.matrikkel.Bruksenhet
+import no.kartverket.matrikkel.bygning.matrikkel.Bygning
+import no.kartverket.matrikkel.bygning.matrikkel.BygningClient
+import no.kartverket.matrikkel.bygning.matrikkelapi.MatrikkelApi
+import no.kartverket.matrikkel.bygning.matrikkelapi.getObjectAs
+import no.kartverket.matrikkel.bygning.matrikkelapi.getObjectsAs
+import no.statkart.matrikkel.matrikkelapi.wsapi.v1.domain.bygning.BygningId
+import no.statkart.matrikkel.matrikkelapi.wsapi.v1.service.store.ServiceException
+import no.statkart.matrikkel.matrikkelapi.wsapi.v1.domain.bygning.Bruksenhet as MatrikkelBruksenhet
+import no.statkart.matrikkel.matrikkelapi.wsapi.v1.domain.bygning.Bygning as MatrikkelBygning
+
+// TODO Håndtering av at matrikkel servicene thrower på visse vanlige HTTP koder, ikke bare full try/catch
+// TODO Logging? Skal vi bruke sl4j for logging i klasser f. eks? Sikkert lurt å ta en runde på logging generelt
+internal class MatrikkelBygningClient(
+    val matrikkelApi: MatrikkelApi.WithAuth
+) : BygningClient {
+    override fun getBygningById(id: Long): Bygning? {
+        val bygningId: BygningId = BygningId().apply { value = id }
+
+        try {
+            val bygning =
+                matrikkelApi.storeService().getObjectAs<MatrikkelBygning>(bygningId, matrikkelApi.matrikkelContext)
+
+            val bruksenheter = matrikkelApi.storeService()
+                .getObjectsAs<MatrikkelBruksenhet>(bygning.bruksenhetIds.item, matrikkelApi.matrikkelContext)
+
+            return Bygning(bygningId = bygning.id.value,
+                bygningNummer = bygning.bygningsnummer,
+                bruksenheter = bruksenheter.map {
+                    Bruksenhet(
+                        bruksenhetId = it.id.value, bygningId = it.byggId.value
+                    )
+                })
+        } catch (exception: ServiceException) {
+            return null
```

Comment:
Per nå bare rethrowe denne så vi i hvert fall kan se feilen i APIet? Eventuelt i hvert fall logge den?

```suggestion
            throw RuntimeException("En feil skjedde under uthenting fra matrikkelen", e)
```

Reply:
Vi må klare å catche 404 på noe vis, bare. Vet ikke om du har noe erfaring med ServiceExceptions som blir kasta av SOAP APIet jeg?

Reply:
Tja, noe. Hovedpoenget var egentlig at det er kjekt å ikke miste all informasjonen fra exceptionen som blir kastet ved å ikke ta vare på den gjennom logging eller rethrowing.
Så fint å i hvert fall logge den hadde vært fint tenker jeg.

Reply:
Ja vil gjerne ha noe logging her. Skal vi sette opp sl4j logging i samme slengen, eller skal jeg bruke noe Ktor greier enn så lenge?

Reply:
Satt det bare opp med KtorSimpleLogger per nå, men virker Ktor default logs har forskjellige håndteringer av logging fra routes kontra andre filer, som jeg ikke er blodfan av.

Tenker vi gjerne kan se på sl4j som felles loggsystem videre

Reply:
Vi bør ha slf4j + logback ja. Ser at `KtorSimpleLogger` i praksis er et alias/en hjelpefunksjon for å opprette en slf4j-logger, så i utgangspunkt ser det ut som man kun trenger å legge inn logback dependency og config som det er beskrevet her: https://ktor.io/docs/server-logging.html

(Og så tenker jeg det er greit å endre til å bruke slf4j direkte i koden).

Reply:
Gønne på med det i denne PRen, bare? Er jo ikke store greiene å legge til @henriwi